compiler/src/dotty/tools/dotc/typer/TypeAssigner.scala

```diff
@@ -475,8 +475,11 @@ trait TypeAssigner {
     assert(!hasNamedArg(args))
     val tparams = tycon.tpe.typeParams
     val ownType =
-      if (sameLength(tparams, args)) tycon.tpe.appliedTo(args.tpes)
-      else wrongNumberOfTypeArgs(tycon.tpe, tparams, args, tree.pos)
+      if (sameLength(tparams, args)) {
+        val argTypes = args.tpes
+        if (argTypes.exists(_ eq NoType)) UnspecifiedErrorType // error reported when typing args
```

Comment:
How do we know the comment is true? I think it is better to defensively issue another error. The worst outcome is an error type in your program without an issued error message.